### DISCUSSION TITLE
Getting changeling into a playable state [transitory document]

### DISCUSSION THREAD

**Discussion by GDNgit** [ParadiseSS13/Paradise#19848] [Documents]
First of all this isn't an offical prog project or something, but it's all stuff that really ought to be done. If you want to do one of these, please contact me on discord, `short name #7913`, because I can probably help you out in some way. 
I have spent a really uncomfortable amount of time designing around this antagonist, and this is something that needs to be talked about.
I want to do a rework on these, but that could take months or even years to properly implement, balance, and get people on board with, not to mention getting someone who has the time to maintain it! So what happens between that time? Well, we do our best to get them into a playable state. 
Below I'm going to detail some things that are, more than likely, needing some tweaks, and some ideas on doing so.
Ordered in 3 sections, Normal contributer stuff (Anyone is welcome to give it a go), Higher difficulty changes (Stuff that may have major coding behind it), And leave this to the experienced (Changes that are extreme or will just innately be hell).

Normal contributer stuff:

- [ ] Add new abilities
New stuff is always cool (And very welcome). Changeling is underdeveloped as all hell, and needs to be given more worthwhile options.

- [ ] Making Panacea no longer use reagents and just do the thing outright #19904
Too difficult to tweak the effectiveness, make it do the thing it says it does in a way that actually makes sense rather than the reagent method now

- [ ] Refactor fleshmend to be more reasonable codewise

- [ ] Rework organic space suit 
Currently this ability is bad... and boring. It encourages the very hated playstyle of spacecling and is objectively worse than just taking fleshmend and using it once in awhile. Give this some reason to be used, or rework it entirely becuase "cheesy flesh suit" is much less terrifying than some normal person who should be dying in space standing still and speedwalking towards you.

- [ ] Make Chitinous armor less awful
This gives you decent armor for the low, low cost of slowdown and -25% chemical regen. Quality ability would get cremated again. Perhaps give this some cool movement tech or a proper shield action considering it's based off Chitin?

- [ ] Make stings easier to use
Stings are kind of a pain to use now and they interfere with a lot of actions. Moving these to a radial menu + having them always be on and just a keystroke sort of thing might be solid. Cutting down on action button bloat is always good

- [ ] Rework Chameleon Skin
This is just a genetic power lol, this needs an identity of it's own.

- [ ] Rework LSD sting
Really should go without saying. Actually griefing tool and adds nothing to the kit, fully replace this if anything

- [ ] Make the shrieks a bit more controlled
For a stealth/ambush based antagonist, a very high range is actually quite the curse. Having no ability to control what exactly you want to go down is bad. A reduction on range would be actually useful and create less spammy strategies.

- [ ] Rework AI untrackable to be a little less bad
It makes you untrackable against the AI and outs you as a changeling. The AI can still see you, just not track you.

- [ ] Make the crematorium either indestructable or able to be fixed. #19878 
The fact this can be destroyed isn't great, if we want to make changelings a little less decap friendly then this MUST stick around.
Best to make this better than the SM dust method as well, don't want people to constantly throw clings the the engine, even if it's quite funny.

Higher difficulty changes:

- [ ] Reworking hivemind based abilities to be less bloat and more useable.
Currently most of the hivemind based abilities are niche at best, and just bloat at worse. Something to make these abilities less awful to use or more reliable would be great. 

- [ ] Rework the evolution point economy 
Evolution points are not in a good spot right now. You get too many of them compared to the cost of the abilities, and you get to take everything you want and more. This means that abilities cannot be balanced around this resource system, as you can just take 5-6 of them. Higher power and more versatile abilities is better than a bunch of low power niche abilities to use and play against.

- [ ] Make Tentacle more consistent and reduce it's power level to encourage more counterplay. #19855
Tentacle is an instant stun provided it works, the ability puts someone into an agressive grab state is crazy and not very fun to play against. What holds Tentacle back is it's buggy nature, which is just awful to play with. 

- [ ] Rework spread infestation into a better ability
Spread infestation isn't fun for anyone. It's either used for spam to spawn a very large amount of spiders for zero reason, or is used to try and cheese people.

- [ ] Make strained muscles and epi-OD more different
Abilities both do speed up but one has antistun. Making them both do different things so that an actual choice needs to be made is good.

- [ ] Make armblade better #19841 
A 25 damage brute weapon just doesn't cut it anymore with newcrit. 

Leave it to the experienced:

- [ ] Increase chemical generation, or otherwise cut down on the amount of waiting while playing changeling #19838 
Waiting isn't a good part of any antagonist, and minimizing it is good. Sitting in a locker for half your round doesn't make people very happy.

- [ ] Radial menu transform
Letting people see what they're transforming into is good. It's also complex and should be done with care.

- [ ] Fixing issues with the transform ability
Species changing is decently buggy, it needs a touch up.

- [ ] Anything relating to species/the mechanics of revive
To be handled with care. Both changes will strike people the wrong way if done improperly.

**Comment by Contrabang** [comment]
Some problems with species changing:
Removes special organs like demon hearts, cursed hearts, etc
Problems with implants not staying intact, but tracker implants staying intact.
Handcuffs being able to be removed by species change
Internals being toggled off on species change (kingstons + N2O is a nightmare).

**Comment by Sirryan2002** [comment]
Alright, I've finally gotten around to reading this (apologies if I ramble, I'm currently on enough cold medicine to knock out a horse).

I've gotten the chance to read both this document (highlighting issues and laying plans) and the TTRPG idea doc (a proposed alternative to current changeling). I'll just bullet-point my thoughts:

* Split the code issues/technical things into their own section; Changeling has a lot of glaring bugs, but if you just refer to a feature as "broken" or in need of a refactor/fix, it does little to point to the actual issue that a coder could bug fix. Furthermore, by differentiating the code issues from the design issues, one can actually start to separate the design problems with changeling from bugs that plague it
* When you refer to "grief sting" I have no idea what you're talking about, could you further define it?
* Be more specific, calling something "bad" is just throwing an adjective at a feature. Unless you plan on reworking every single ability yourself, describe in detail what's wrong with it.

I like when you were specific with a few points. Namely, talking about what you want seen done with Anatomic Panacea and making stings easier to use. These are following a good design thinking track of `Identifiy Issue` -> `Discuss` -> `Propose Solution` which I think can be more implemented everywhere in your doc. 

Try and combine what you're doing here with the approach you took to your other doc, tie the analysis and commentary on each specific ability with a vision of how a changeling player will progress throughout an entire round. What ability should cost what and at what point do you think players should be unlocking each ability throughout the round? 

I think you could really do with analyzing the gamemode as a whole and breaking down what you think the progression of it is now, the dominant strategies employed, and how the current features work with/against changeling players in relation to the round (such as security or changeling targets). Right now its hard to give any binding commentary on the proposed changes because the suggestions are relatively surface-level.

**Reply by GDNgit** [reply]
> * Split the code issues/technical things into their own section; Changeling has a lot of glaring bugs, but if you just refer to a feature as "broken" or in need of a refactor/fix, it does little to point to the actual issue that a coder could bug fix. Furthermore, by differentiating the code issues from the design issues, one can actually start to separate the design problems with changeling from bugs that plague it

Only transform needs bug fixes, everything else is just a bit weird design wise, having a whole section on that wouldn't really fit this type of document.

> * When you refer to "grief sting" I have no idea what you're talking about, could you further define it?

LSD sting, RNG based kick in timer, extremely spam-able. Universally hated. 

> * Be more specific, calling something "bad" is just throwing an adjective at a feature. Unless you plan on reworking every single ability yourself, describe in detail what's wrong with it.

Transition documents are supposed to be less specific and with an antagonist like changeling a 1 on 1 discussion is going to be much better for both parties rather than some blanket statement on why something doesn't work. Everyone has different feelings on how something should be balanced, why it's an issue, and how to solve it. 

> I like when you were specific with a few points. Namely, talking about what you want seen done with Anatomic Panacea and making stings easier to use. These are following a good design thinking track of `Identifiy Issue` -> `Discuss` -> `Propose Solution` which I think can be more implemented everywhere in your doc.

Transition document, not supposed to be super specific. I can give people ideas if THEY ask for them, but I am trying to avoid that to avoid people getting demotivated. I want to see people tap into their own creativity rather than having a project go nowhere.

> Try and combine what you're doing here with the approach you took to your other doc, tie the analysis and commentary on each specific ability with a vision of how a changeling player will progress throughout an entire round. What ability should cost what and at what point do you think players should be unlocking each ability throughout the round?

I'm not sure where this came from, but I have zero personal interest in progression of any kind for changeling. We already have a good progression antagonist in vampire.

> I think you could really do with analyzing the gamemode as a whole and breaking down what you think the progression of it is now, the dominant strategies employed, and how the current features work with/against changeling players in relation to the round (such as security or changeling targets). 

I can do this but not in this document, I also don't want to demotivate people by being a downer, especially now where burnout is at its worse for a lot of people. 

> Right now its hard to give any binding commentary on the proposed changes because the suggestions are relatively surface-level.

I just wanted to open some dialogue with what exactly people want and have an open discussion on what is fun or not, I don't see the value in anything being binding when this should be a community wide project. If we make stuff binding that's just gonna make people burn out even harder. 
Also hope you feel better soon Ryan, and anyone here is totally welcome to just ping or DM whatever and we can talk about adding new abilities/reworking them.
